File: src/front-end/src/app/services/task.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { TodoTask } from '../entity/todo-task';
import { environment } from 'src/environments/environment';
import { User } from '../entity/user';

@Injectable({
  providedIn: 'root',
})
export class TaskService {
  private url = environment.backendUrl + '/tasks';
  storage: Storage = sessionStorage;

  constructor(private http: HttpClient) {}

  getTasks(user: User) {
    const getUserTasksUrl = this.url + `/findByUser?userId=${user.id}`;
    return this.http.get<TodoTask[]>(getUserTasksUrl);
  }

  updateTask(task: TodoTask) {
    return this.http.put<TodoTask>(this.url, task);
  }

  createTask(task: TodoTask) {
    return this.http.post<TodoTask>(this.url, task);
  }
  deleteTask(task: TodoTask) {
    return this.http.delete(this.url + `/${task.id}`);
  }
}

File: src/main/java/me/todoapp/service/TaskService.java
package me.todoapp.service;

import me.todoapp.entity.Task;
import me.todoapp.entity.User;

import java.util.List;

public interface TaskService {

    List<Task> getTest();

    Task createTask(Task task);

    List<Task> getAllTasks(String userId);


    Task getTask(Long id);

    Task updateTask(Task task);

    void deleteTask(Long taskId);

    Task completeTask(Task task);
}
